musicbot/billboard/Song.py

```diff
@@ -0,0 +1,105 @@
+import discord
+import json
+import logging
+import os
+from pathlib import Path
+
+from musicbot.lib.JsonFile import JsonFile
+
+
+log = logging.getLogger(__name__)
+
+class Song:
+    def __init__(self, video_information_dictionary, working_directory):
+        log.debug("Creating Song. video_information_dictionary: {}, working_directory: {}".format(str(video_information_dictionary), str(working_directory)))
+        self._base_working_directory = working_directory
+        self._video_id = video_information_dictionary["video_id"]
+        self._working_directory = self._base_working_directory + self._video_id + "\\"
+        self._info_json_path = self._working_directory + "info.json"
+
+        Path(self._working_directory).mkdir(exist_ok=True)
+        self._initializeInfoJson(video_information_dictionary)
+      
+    def _initializeTitle(self, loaded_info_json, default_title):
+        key = "title"
+        title = "null" if key not in loaded_info_json else loaded_info_json[key]
+        self._info_json[key] = title if title != "null" else default_title
+        
+    def _initializeKeyWithDefaultValue(self, key, loaded_info_json):
+        self._info_json[key] = self._info_json[key] if key not in loaded_info_json else loaded_info_json[key]
+         
+    def _initializeInfoJson(self, video_information_dictionary):
+        log.debug("_initializeInfoJson called. video_information_dictionary: {}".format(str(video_information_dictionary)))
+        self._info_json = {
+            "title" : "null",   
+            "times_queued" : 0,
+            "number_of_weeks_on_chart" : -1,
+            "position_this_week" : -1,
+            "position_last_week" : -1,
+            "peak_position" : -1 
+        }
```

Comment:
I thought this was a reference of the json parser as to what values are to be decoded. But after reading more code, it seems that this is used for filling the initial values of a new song. I suggest to make it more visible/readable

Reply:
done. created class variable